[3.4.2] Docker microservices - Compatibility with Compose V2

trikimiki opened the pull request:
## Pull Request description

Docker microservices installation (aka Docker Compose installation) have been using an outdated Compose V1. As Docker no longer support V1 and there are differences between V1 and V2, some script and compose file changes were required.  

- remove `version` tag, as it is deprecated in V2;
- change `scale` to `deploy.replicas` due to deprecation and weird interactions with multiple "-f" flags;   
- change `docker-compose` to `docker compose` that corresponds to migration to V2.

Documentation changes are not required, as Docker instructions already use Compose V2 only.

Issue: #7685 

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] PR name contains fix version. For example, "[3.3.4] Hotfix of some UI component" or "[3.4] New Super Feature".
- [x] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Required for internal contributors only.
  
## Front-End feature checklist

- [ ] Screenshots with affected component(s) are added. The best option is to provide 2 screens: before and after changes;
- [ ] If you change the widget or other API, ensure it is backward-compatible or upgrade script is present.
- [ ] Ensure new API is documented [here](https://github.com/thingsboard/thingsboard-ui-help)

## Back-End feature checklist

- [ ] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [ ] If new dependency was added: the dependency tree is checked for conflicts.
- [ ] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [ ] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.

volodymyr-babak commented:
@trikimiki 
Is it still possible to run ThingsBoard using docker-compose? 
Or user is forced to switch to Docker Compose V2?

trikimiki commented:
User is forced to switch to Compose V2.
Still, according to the upgrade guide, user does not need to merge script changes, and ThingsBoard will start/stop/upgrade/etc with old scripts as intended.

trikimiki commented:
TODO: add an option for compatibility with Compose V1

trikimiki commented:
Also added backward compatibility for Compose V1. Refactored install/start/stop/etc Bash scripts along the way.

- [x] Changes are backward compatible now
- [x] Code was reviewed by @volodymyr-babak
